backend/SlothOrganizer/SlothOrganizer.Persistence/Repositories/TaskCompletionRepository.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Threading.Tasks;
+using Dapper;
+using SlothOrganizer.Domain.Entities;
+using SlothOrganizer.Domain.Repositories;
+using SlothOrganizer.Persistence.Properties;
+
+namespace SlothOrganizer.Persistence.Repositories
+{
+    public class TaskCompletionRepository : ITaskCompletionRepository
+    {
+        private readonly DapperContext _context;
+
+        public TaskCompletionRepository(DapperContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<List<TaskCompletion>> Insert(List<TaskCompletion> taskCompletions)
+        {
+            var query = Resources.InsertTaskCompletion;
+
+            using var connection = _context.CreateConnection();
+            foreach (var taskCompletion in taskCompletions)
```

Comment:
I guess it may be rewritten with running queries simultaneously with usage Tas.WhenAll instead of running one by one.

Reply:
Database connection isn't thread-safe, so to run insertions in parallel we have to open and close connection for each insertion separately. Because of it running insertions synchronously is actually faster

Reply:
I guess even it is not a thread safe you may use a transaction and perform everything with a single request.

Reply:
Fixed to execute all insertions in a single transaction